components/backup/src/endpoint.rs

```diff
@@ -913,23 +1020,19 @@ impl<E: Engine, R: RegionInfoProvider + Clone + 'static> Endpoint<E, R> {
     pub fn handle_backup_task(&self, task: Task) {
         let Task { request, resp } = task;
         let is_raw_kv = request.is_raw_kv;
-        let start_key = if request.start_key.is_empty() {
-            None
-        } else {
-            // TODO: if is_raw_kv is written everywhere. It need to be simplified.
-            if is_raw_kv {
-                Some(Key::from_encoded(request.start_key.clone()))
-            } else {
-                Some(Key::from_raw(&request.start_key))
+        if is_raw_kv
```

Comment:
Suggest to encapsulate `is_raw_kv`, `api_version`, `dst_api_version`, and related methods.

Reply:
done